extensions/amp-analytics/0.1/visibility-model.js

```diff
@@ -51,6 +51,17 @@ export class VisibilityModel {
       this.spec_['visiblePercentageMax'] = 0;
     }
 
+    /** @private @const {boolean} */
+    this.accumulateCounters_ = spec['reportWhen'] !== undefined;
+
+    /**
+     * Allow zero visible time on screen when explicitly provided and when the
+     * ping is deferred to some other event.
+     * @private @const {boolean}
+     */
+    this.forceReport_ =
```

Comment:
We need to make sure `hidden` trigger, `reportWhen`, `forceReport` and `repeat` work well together. 
Right now: `hidden` trigger equals to `reportWhen: pageHidden`. `reportWhen` doesn't work with `repeat` with the special case, `hidden` trigger works with `repeat`. 

We can be more strict on checking `forceReport` only works with `reportWhen` but not `repeat: true`. One approach is to have `reportWhen` be an object and have `forceReport` be one of its field. But there's the tradeoff making the configuration too complicated. 

WDYT? cc @lannka

Reply:
I'm not fully familiar with the behavior of `repeat`. Will it force a report?

In this case, it'd be possible to get force multiple pings if the `reportWhen` fires multiple times. (Which, to confirm, can happen with the `hidden` trigger as it's based on `visibilitychange` events?) This seems like possibly the right behavior, e.g. to implement a heartbeat ping. Though maybe we want to toggle `forceReport` on repeated triggers to prevent users from implementing heartbeat pings?

Reply:
Not an issue any more given we are going to change the `reportWhen` default behavior.